### PR TITLE
[#17] [iOS] [Integrate] As a user, I can see a splash screen when app starts

### LINKED ISSUE #17: [iOS] [Integrate] As a user, I can see a splash screen when app starts
# Why

When the application starts, the splash screen will be shown to keep users occupied while the application is fetching necessary data.

# Acceptance Criteria
- Show the splash screen when the app starts.
- Check if the user is logged in.
- Refresh their access token if it is expired. (backend story #9)
- Save the new access token for future usage.

### PR DESCRIPTION
**Pull request by phongvhd93**
- Close #17

## What happened 👀

- Add KSP task and mockative library to generate mock objects.
- Write unit tests for `SplashViewModel`, `AuthenticationRepository`, `CheckLoggedInUseCase`
- Write UI test for `SplashView`

## Insight 📝

Integrate Splash View to the login app flow and write test example

## Proof Of Work 📹

https://user-images.githubusercontent.com/22606906/233891997-c36e9d04-1e56-4541-9bb9-89c84eb39126.mp4

### CODE REVIEW

File: iosApp/PhongKMMIC/Sources/Presentation/Modules/Splash/SplashView.swift [hunk 1]
```diff
@@ -6,28 +6,36 @@
 //  Copyright © 2023 Nimble. All rights reserved.
 //
 
+import shared
 import SwiftUI
 
 struct SplashView: View {
 
-    @State private var isLoaded = false
+    @StateObject private var viewModel: SplashViewModel = KoinApplication.inject(\.splashViewModel)
```
Comment (suho) [inline iosApp/PhongKMMIC/Sources/Presentation/Modules/Splash/SplashView.swift]: Please consider to use `Factory`

File: iosApp/fastlane/Constants/Constant.swift [hunk 2]
```diff
@@ -18,7 +18,7 @@ enum Constant {
 
     static let stagingFirebaseAppId = "1:1005614911399:ios:8051ebc71d585a2abe3082"
     static let productionFirebaseAppId = "1:1005614911399:ios:0442661faf675297be3082"
-    static let firebaseTesterGroups = "" //"group_name, group_name"
+    static let firebaseTesterGroups = "" // "group_name, group_name"
```
Comment (suho) [inline iosApp/fastlane/Constants/Constant.swift]: Please revert the change

File: iosApp/PhongKMMIC/Sources/Application/AppDelegate.swift [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+import UIKit
+
+final class AppDelegate: NSObject, UIApplicationDelegate {
+
+    func application(
+        _ application: UIApplication,
+        didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]? = nil
+    ) -> Bool {
+        UITestHelper.shared.speedUpUITestAnimation()
```
Comment (markgravity) [inline iosApp/PhongKMMIC/Sources/Application/AppDelegate.swift]: What do you think about wrapping it with `#if DEBUG`?
It's more safety in the Production 🙏

File: iosApp/PhongKMMIC/Sources/Supports/Extensions/KMM/ViewModels/SplashViewModel+ObservableObject.swift [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+//  SplashViewModel+ObservableObject.swift
+//  PhongKMMIC
+//
+//  Created by Phong Vo on 20/04/2023.
+//  Copyright © 2023 Nimble. All rights reserved.
+//
+
+import shared
+
+extension SplashViewModel: ObservableObject {}
```
Comment (markgravity) [inline iosApp/PhongKMMIC/Sources/Supports/Extensions/KMM/ViewModels/SplashViewModel+ObservableObject.swift:11]: I don't think putting the ViewModel into the Shared module is a good idea, what's happened if we need to use Combine in this screen 🤔

Comment (phongvhd93) [reply iosApp/PhongKMMIC/Sources/Supports/Extensions/KMM/ViewModels/SplashViewModel+ObservableObject.swift:11]: Maybe a wrapper like Bliss did. Or we can transform Kotlin Flow to Combine, there's many resources about that, and I want to validate them.

File: iosApp/PhongKMMIC/Sources/Application/AppDelegate.swift (hunk 3 above)
Comment (phongvhd93) [reply iosApp/PhongKMMIC/Sources/Application/AppDelegate.swift]: Nice catch, I'll fix that.

Comment (phongvhd93) [reply iosApp/PhongKMMIC/Sources/Application/AppDelegate.swift]: solved in 848e473

File: iosApp/PhongKMMIC/Sources/Supports/Helpers/UITests/UITestHelper.swift [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    static let shared = UITestHelper()
+
+    var speed: Float = 1.0
+
+    var isFastAnimationUITests: Bool {
+        LaunchArgument.contains(.fastAnimation)
+    }
+
+    func speedUpUITestAnimation() {
+        #if DEBUG
```
Comment (markgravity) [inline iosApp/PhongKMMIC/Sources/Supports/Helpers/UITests/UITestHelper.swift:22]: By doing this
```
 #if DEBUG
    UITestHelper.shared.speedUpUITestAnimation()
#endif
```
I think we don't need to use `#if DEBUG` in here anymore, what do you think?

**Review by markgravity — APPROVED** [review]
LGTM with a small optional suggestion

File: iosApp/PhongKMMIC/Sources/Supports/Helpers/UITests/UITestHelper.swift (hunk 5 above)
Comment (phongvhd93) [reply iosApp/PhongKMMIC/Sources/Supports/Helpers/UITests/UITestHelper.swift:22]: I think it is better to keep that in `AppDelegate` it easier to read and immediately acknowledge that function only run on DEBUG

**Review by suho — APPROVED** [review]
